nuxeo-runtime/nuxeo-launcher-commons/src/main/java/org/nuxeo/launcher/config/ConfigurationGenerator.java

```diff
@@ -1272,29 +1281,65 @@ public void verifyInstallation() throws ConfigurationException {
      * @since 5.6
      */
     public void checkJavaVersion() throws ConfigurationException {
+        String requiredVersion = COMPLIANT_JAVA_VERSIONS[0];
         String version = System.getProperty("java.version");
-        boolean isCompliant = false;
-        boolean isGreater = false;
-        for (String compliantJava : COMPLIANT_JAVA_VERSIONS) {
-            if (version.startsWith(compliantJava)) {
-                isCompliant = true;
-                break;
-            } else if (version.compareTo(compliantJava) > 0) {
-                isGreater = true;
-            }
+        if (!checkJavaVersion(version, requiredVersion, true, true)) {
+            String message = String.format("Nuxeo requires Java %s+ (detected %s).", requiredVersion, version);
+            throw new ConfigurationException(message + " See '" + JVMCHECK_PROP + "' option to bypass version check.");
         }
-        if (!isCompliant) {
-            String message = String.format(
-                    "Nuxeo requires Java %s (detected %s).",
-                    ArrayUtils.toString(COMPLIANT_JAVA_VERSIONS), version);
-            if (isGreater
-                    || "nofail".equalsIgnoreCase(System.getProperty("jvmcheck",
-                            "fail"))) {
-                log.warn(message);
-            } else {
-                throw new ConfigurationException(message);
+    }
+
+    /**
+     * Checks the java version compared to the required one.
+     * <p>
+     * Loose compliance is assumed if the major version is greater than the required major version or a jvmcheck=nofail
+     * flag is set.
+     *
+     * @param version the java version
+     * @param requiredVersion the required java version
+     * @param allowNoFailFlag if {@code true} then check jvmcheck=nofail flag to always have loose compliance
+     * @param warnIfLooseComliance if {@code true} then log a WARN if the is loose compliance
+     * @return true if the java version is compliant (maybe loosely) with the required version
+     * @since 8.4
+     */
+    protected static boolean checkJavaVersion(String version, String requiredVersion, boolean allowNoFailFlag,
+            boolean warnIfLooseComliance) {
+        String[] versionSplit = version.split("_");
+        String[] requiredVersionSplit = requiredVersion.split("_");
+        String major = versionSplit[0];
+        String requiredMajor = requiredVersionSplit[0];
+        int minor = Integer.parseInt(versionSplit[1]);
+        int requiredMinor = Integer.parseInt(requiredVersionSplit[1]);
```

Comment:
According to http://www.oracle.com/technetwork/java/javase/versioning-naming-139433.html `java.version`may be something like `1.8.0_32-ea`. These are not GA java release but we could at least catch a `NumberFormatException` here (return `false` plus a log saying we've not been able to parse version)